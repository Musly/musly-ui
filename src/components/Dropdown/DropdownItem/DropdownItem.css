.dropdown__item__link {
  display: flex;
  align-items: center;
  transition-property: color, background-color;
  transition-duration: 200ms;

  @apply py-3 px-4 bg-transparent ease-sharp text-gray-800;

  &--active {
    font-weight: 600;
    cursor: default;
    pointer-events: none;

    .dropdown__icon {
      @apply text-blue-500;
    }
  }

  &:not(.dropdown__item__link--active):hover {
    @apply bg-gray-100 text-blue-500;
  }
}

.dropdown__icon {
  @apply mr-4 text-gray-500;
}
