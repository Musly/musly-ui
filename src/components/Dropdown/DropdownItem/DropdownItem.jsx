import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './DropdownItem.css';

export function DropdownItem (props) {
  const {
    url, label, icon, active, ...rest
  } = props;

  const linkClasses = classNames({
    dropdown__item__link: true,
    'dropdown__item__link--active': active,
  });

  return (
    <li className="dropdown__item" role="presentation">
      <a href={url} {...rest} role="menuitem" className={linkClasses}>
        {icon && (
          <div className="dropdown__icon" data-test-id="icon">
            {icon}
          </div>
        )}
        <span className="flex-grow" data-test-id="label">
          {label}
        </span>
      </a>
    </li>
  );
}

DropdownItem.propTypes = {
  url: PropTypes.string,
  label: PropTypes.string.isRequired,
  icon: PropTypes.node,
  active: PropTypes.bool,
};

DropdownItem.defaultProps = {
  url: '/',
  icon: null,
  active: false,
};
