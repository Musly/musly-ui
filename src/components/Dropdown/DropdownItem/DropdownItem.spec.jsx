import React from 'react';
import { shallow } from 'enzyme';
import HomeIcon from '../../../icons/home.svg';
import { DropdownItem } from './DropdownItem';

describe('<DropdownItem />', () => {
  it('should render without icon', () => {
    const result = shallow(<DropdownItem url="/test" label="Test Label" />);
    expect(result.find('a').props().href).toEqual('/test');
    expect(result.find('[data-test-id="icon"]')).not.toExist();
    expect(result.find('[data-test-id="label"]').props().children).toEqual('Test Label');
    expect(result).toMatchSnapshot();
  });

  it('should render with icon', () => {
    const result = shallow(<DropdownItem url="/test" label="Test Label" icon={<HomeIcon />} />);
    expect(result.find('[data-test-id="icon"]')).toExist();
    expect(result).toMatchSnapshot();
  });
});
