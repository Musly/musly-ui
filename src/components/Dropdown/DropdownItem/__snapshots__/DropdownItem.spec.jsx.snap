// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownItem /> should render with icon 1`] = `
<li
  className="dropdown__item"
  role="presentation"
>
  <a
    className="dropdown__item__link"
    href="/test"
    role="menuitem"
  >
    <div
      className="dropdown__icon"
      data-test-id="icon"
    >
      <home.svg />
    </div>
    <span
      className="flex-grow"
      data-test-id="label"
    >
      Test Label
    </span>
  </a>
</li>
`;

exports[`<DropdownItem /> should render without icon 1`] = `
<li
  className="dropdown__item"
  role="presentation"
>
  <a
    className="dropdown__item__link"
    href="/test"
    role="menuitem"
  >
    <span
      className="flex-grow"
      data-test-id="label"
    >
      Test Label
    </span>
  </a>
</li>
`;
